src/Microsoft.FeatureManagement/ConfigurationFeatureDefinitionProvider.cs

```diff
@@ -113,16 +117,58 @@ public async IAsyncEnumerable<FeatureDefinition> GetAllFeatureDefinitionsAsync()
             // Iterate over all features registered in the system at initial invocation time
             foreach (IConfigurationSection featureSection in GetFeatureDefinitionSections())
             {
+                string featureName = GetFeatureFlagSectionName(featureSection);
+
+                if (string.IsNullOrEmpty(featureName))
+                {
+                    continue;
+                }
+
                 //
                 // Underlying IConfigurationSection data is dynamic so latest feature definitions are returned
-                yield return  _definitions.GetOrAdd(featureSection.Key, (_) => ReadFeatureDefinition(featureSection));
+                yield return  _definitions.GetOrAdd(featureName, (_) => ReadFeatureDefinition(featureSection));
+            }
+        }
+
+        private void EnsureInit()
+        {
+            if (!_configuration.GetChildren().Any())
+            {
+                Logger?.LogDebug($"Configuration is empty.");
+            }
+
+            bool hasFeatureManagementSection = _configuration.GetChildren()
+                .Any(section => string.Equals(section.Key, ConfigurationFields.FeatureManagementSectionName, StringComparison.OrdinalIgnoreCase));
+
+            if (!hasFeatureManagementSection && !RootConfigurationFallbackEnabled)
+            {
+                Logger?.LogDebug($"No configuration section named '{ConfigurationFields.FeatureManagementSectionName}' was found.");
+            }
+
+            if (_initialized == 0)
+            {
+                IConfiguration featureManagementConfigurationSection = hasFeatureManagementSection ?
+                    _configuration.GetSection(ConfigurationFields.FeatureManagementSectionName) :
+                    _configuration;
```

Comment:
Why is there fallback to the root configuration without a check for `RootConfigurationFallbackEnabled`

Reply:
Perhaps this used to return above previously. It needs to be corrected given the change.